### DISCUSSION TITLE
Windows Hello biometrics login integratiion

### DISCUSSION THREAD

**Discussion by VShawn** [1Remote/1Remote#429] [Ideas]
It was alway someone ask me for a password protection for the APP, and it's annoying for me to enter passwords every time opening the app.

But a fingerprint recognizer or face recognizer would be nice for the app unlock.

I‘m doubting if we can integrate Windows Hello biometrics login function in to PRemoteM. 

Does anyone give  some advices?

refacing: https://docs.microsoft.com/en-us/windows/uwp/security/microsoft-passport-login

**Comment by majkinetor** [comment]
One advice is to cache password for some time like SSH agent does (configurable, such as 1 hour). It could expire it ASAP if OS is locked. 

Its still annoying tho. Fingerprint recognizer doesn't strike me as great, its only there on laptops and only on some of them. Camera seems best ubiquitous solution, so Windows Hello really sounds interesting. I have 0 experience with enabling that on random apps nor I have seen any doing that (I have seen it on Android phones).

**Comment by majkinetor** [comment]
I doubt this would be used by many users in any case.

**Comment by VShawn** [comment]
I found the approach on personal password manage app

https://github.com/sirAndros/KeePassWinHello

**Comment by VShawn** [comment]
after some test, I found that windows hello can be referenced in appx packed app. This means only the store version can support the windows hello.

**Comment by majkinetor** [comment]
Cant find that this is strict requirement, any links ?

**Comment by VShawn** [comment]
I'm not very sure. Because there no much windows hello doc for desktop app, we have to investigate by ourselves.

Since windwos hello required KeyCredentialManager in Windows 10 APIs and most of these APIs require an `identity`

Then doc says `You can easily create an identity for your WPF app by packaging it as MSIX. `

So the portable app may have no ability to call some Windows 10 APIs included the KeyCredentialManager
